x-pack/plugins/security_solution/public/timelines/components/side_panel/network_details/index.tsx

```diff
@@ -0,0 +1,209 @@
+import { useNetworkDetails } from '../../../../network/containers/details';
+import { networkModel } from '../../../../network/store';
+import { useAnomaliesTableData } from '../../../../common/components/ml/anomaly/use_anomalies_table_data';
+
+const StyledEuiFlyoutBody = styled(EuiFlyoutBody)`
+  .euiFlyoutBody__overflow {
+    display: flex;
+    flex: 1;
+    overflow-x: hidden;
+    overflow-y: scroll;
+
+    .euiFlyoutBody__overflowContent {
+      flex: 1;
+      overflow-x: hidden;
+      overflow-y: scroll;
+      padding: ${({ theme }) => `${theme.eui.paddingSizes.xs} ${theme.eui.paddingSizes.m} 64px`};
+    }
+  }
+`;
+
+const StyledEuiFlexGroup = styled(EuiFlexGroup)`
+  flex: 0;
+`;
+
+const StyledEuiFlexItem = styled(EuiFlexItem)`
+  &.euiFlexItem {
+    flex: 1 0 0;
+    overflow-y: scroll;
+    overflow-x: hidden;
+  }
+`;
+
+const StyledEuiFlexButtonWrapper = styled(EuiFlexItem)`
+  align-self: flex-start;
+`;
+
+interface NetworkDetailsProps {
+  expandedNetwork: { ip: string; flowTarget: FlowTarget };
+  handleOnNetworkClosed: () => void;
+  isFlyoutView?: boolean;
+}
+
+export const NetworkDetailsPanel = React.memo(
+  ({ expandedNetwork, handleOnNetworkClosed, isFlyoutView }: NetworkDetailsProps) => {
+    const { ip, flowTarget } = expandedNetwork;
+    const dispatch = useDispatch();
+    const { to, from, isInitializing } = useGlobalTime();
+    const getGlobalQuerySelector = useMemo(() => inputsSelectors.globalQuerySelector(), []);
+    const getGlobalFiltersQuerySelector = useMemo(
+      () => inputsSelectors.globalFiltersQuerySelector(),
+      []
+    );
+
+    const query = useDeepEqualSelector(getGlobalQuerySelector);
+    const filters = useDeepEqualSelector(getGlobalFiltersQuerySelector);
+
+    const type = networkModel.NetworkType.details;
+    const narrowDateRange = useCallback(
+      (score, interval) => {
+        const fromTo = scoreIntervalToDateTime(score, interval);
+        dispatch(
+          setAbsoluteRangeDatePicker({
+            id: 'global',
+            from: fromTo.from,
+            to: fromTo.to,
+          })
+        );
+      },
+      [dispatch]
+    );
+    const {
+      services: { uiSettings },
+    } = useKibana();
+
+    const { docValueFields, indicesExist, indexPattern, selectedPatterns } = useSourcererScope();
+    const filterQuery = convertToBuildEsQuery({
+      config: esQuery.getEsQueryConfig(uiSettings),
+      indexPattern,
+      queries: [query],
+      filters,
+    });
+
+    const [loading, { id, networkDetails }] = useNetworkDetails({
+      docValueFields,
+      skip: isInitializing,
+      filterQuery,
+      indexNames: selectedPatterns,
+      ip,
+    });
+
+    const [isLoadingAnomaliesData, anomaliesData] = useAnomaliesTableData({
+      criteriaFields: networkToCriteria(ip, flowTarget),
+      startDate: from,
+      endDate: to,
+      skip: isInitializing,
+    });
+
+    const ExpandableNetworkWrapper = () =>
```

Comment:
same as above, let's not use function inside of the component

Reply:
Yea. I created separate components for all of these